pkg/controllers/cloud_config_sync_controller.go

```diff
@@ -183,27 +177,6 @@ func (r *CloudConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
-func (r *CloudConfigReconciler) isCloudConfigSyncNeeded(platformStatus *configv1.PlatformStatus, infraCloudConfigRef configv1.ConfigMapFileReference) (bool, error) {
-	if platformStatus == nil {
-		return false, fmt.Errorf("platformStatus is required")
-	}
-	switch platformStatus.Type {
-	case configv1.AzurePlatformType,
-		configv1.GCPPlatformType,
-		configv1.VSpherePlatformType,
-		configv1.IBMCloudPlatformType,
-		configv1.PowerVSPlatformType,
-		configv1.OpenStackPlatformType,
-		configv1.NutanixPlatformType:
```

Comment:
Does removing this mean that we will now sync a metal platform configmap? Or does the sync controller just not run at all on metal?

Reply:
It does run on metal, and removing this caused their tests to fail. I've added it back and simply moved AWS's entry up. This way we're not changing behavior on platforms that relied on it before.